spring-grpc-core/src/main/java/org/springframework/grpc/client/GrpcChannelBuilder.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.grpc.client;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.function.Function;
+
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingChannelBuilder2;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+/**
+ * An extended {@link ManagedChannelBuilder} that allows setting the shutdown grace period
+ * for created channels as well as more options for client interceptors.
+ *
+ * @author Chris Bono
+ */
+public class GrpcChannelBuilder extends ForwardingChannelBuilder2<GrpcChannelBuilder> {
```

Comment:
Just like the NettyServerBuilder provides extended functionality, we do the same here for:

1. Add shutdown grace period
2. Add merge interceptors indicator
3. Override existing intercept method append or merge w/ global interceptors depending on above indicators (NYI but it shows the contract and how we can do it)

Reply:
I wanted to avoid this as it's so close to the native `ManagedChannelBuilder`, but I think you have proved it is probably necessary.